cloudflare/resource_cloudflare_access_mutual_tls_certificate.go

```diff
@@ -0,0 +1,195 @@
+
+	log.Printf("[DEBUG] Creating Cloudflare Access Mutual TLS certificate from struct: %+v", newAccessMutualTLSCertificate)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	var accessMutualTLSCert cloudflare.AccessMutualTLSCertificate
+	if identifier.Type == AccountType {
+		accessMutualTLSCert, err = client.CreateAccessMutualTLSCertificate(identifier.Value, newAccessMutualTLSCertificate)
+	} else {
+		accessMutualTLSCert, err = client.CreateZoneAccessMutualTLSCertificate(identifier.Value, newAccessMutualTLSCertificate)
+	}
+	if err != nil {
+		return fmt.Errorf("error creating Access Mutual TLS Certificate for %s %q: %s", identifier.Type, identifier.Value, err)
+	}
+
+	d.SetId(accessMutualTLSCert.ID)
+
+	return resourceCloudflareAccessMutualTLSCertificateRead(d, meta)
+}
+
+func resourceCloudflareAccessMutualTLSCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	var accessMutualTLSCert cloudflare.AccessMutualTLSCertificate
+	if identifier.Type == AccountType {
+		accessMutualTLSCert, err = client.AccessMutualTLSCertificate(identifier.Value, d.Id())
+	} else {
+		accessMutualTLSCert, err = client.ZoneAccessMutualTLSCertificate(identifier.Value, d.Id())
+	}
+
+	if err != nil {
+		if strings.Contains(err.Error(), "HTTP status 404") {
+			log.Printf("[INFO] Access Mutal TLS Certificate %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error finding Access Mutual TLS Certificate %q: %s", d.Id(), err)
+	}
+
+	d.Set("name", accessMutualTLSCert.Name)
+	d.Set("associated_hostnames", accessMutualTLSCert.AssociatedHostnames)
+	d.Set("fingerprint", accessMutualTLSCert.Fingerprint)
+
+	return nil
+}
+
+func resourceCloudflareAccessMutualTLSCertificateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	updatedAccessMutualTLSCert := cloudflare.AccessMutualTLSCertificate{
+		ID:   d.Id(),
+		Name: d.Get("name").(string),
+	}
+	updatedAccessMutualTLSCert.AssociatedHostnames = expandInterfaceToStringList(d.Get("associated_hostnames"))
+
+	log.Printf("[DEBUG] Updating Cloudflare Access Mutal TLS Certificate from struct: %+v", updatedAccessMutualTLSCert)
+
+	identifier, err := initIdentifier(d)
+	if err != nil {
+		return err
+	}
+
+	var accessMutualTLSCert cloudflare.AccessMutualTLSCertificate
+	if identifier.Type == AccountType {
+		accessMutualTLSCert, err = client.UpdateAccessMutualTLSCertificate(identifier.Value, d.Id(), updatedAccessMutualTLSCert)
+	} else {
+		accessMutualTLSCert, err = client.UpdateZoneAccessMutualTLSCertificate(identifier.Value, d.Id(), updatedAccessMutualTLSCert)
+	}
+	if err != nil {
+		return fmt.Errorf("error updating Access Mutual TLS Certificate for %s %q: %s", identifier.Type, identifier.Value, err)
+	}
+
+	if accessMutualTLSCert.ID == "" {
+
+	}
```

Comment:
i don't _think_ we need this for anything in the `Update`

Reply:
Ah, I think I forgot to clean this up. I noticed we did this in a few places such as https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_access_application.go#L278-L280 and https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_access_policy.go#L177-L179 but I don't think it's an actual concern.